unittests/vulkantests.py

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2016-2025 The Meson development team
+
+import tempfile
+import textwrap
+from pathlib import Path
+from unittest import mock
+
+from .baseplatformtests import BasePlatformTests
+from mesonbuild.compilers.mixins.clike import CLikeCompiler
+
+
+class VulkanTests(BasePlatformTests):
+
+    def test_vulkan_sdk_environment_variable(self):
+        '''
+        Test that the SDK version pointed to by the VULKAN_SDK
+        environment variable is found.
+        In this test we're only interested in finding the Vulkan
+        version. Finding the library is mocked.
+        '''
```

Comment:
This seems quite, quite complex, and shouldn't be.

We already have `test cases/frameworks/18 vulkan/`

- We definitely should not be generating a new meson.build in a tmpdir for this
- Does this existing test not already fail when the whole testsuite is run and `$VULKAN_SDK` is set?
- Adding a whole new unittests category seems overkill, and it's not actually hooked up?

Reply:
> Does this existing test not already fail when the whole testsuite is run and $VULKAN_SDK is set?

The existing test just assumes Vulkan is not installed and skips the test.

> Adding a whole new unittests category seems overkill, and it's not actually hooked up?

I'm happy to remove the test case if you think it's not needed.

Not hooked up? Isn't that what Pytest is supposed to do?

Reply:
pytest is optional and used if available. If it is not available, the stdlib unittest is used.

https://github.com/mesonbuild/meson/blob/c98cd76956a7ece7f5eea41407651e0af1d37152/run_unittests.py#L137-L141

I'm not saying the unittest must be removed -- I'm saying we should probably expand the existing one.

Please also see https://mesonbuild.com/Contributing.html#configuring-project-tests for how to set an environment variable, including how to use `@ROOT@` to set it to values relative to the test source directory.

Reply:
I do not think a project test will work, unless there a host where the Vulkan SDK is installed in a non-default location, so we can use the $VULKAN_SDK env var. Vulkan should not be resolved by pkg-config.

Alternatively it may be an idea to create a unit test that targets `VulkanDependencySystem` directly.

Reply:
> Alternatively it may be an idea to create a unit test that targets `VulkanDependencySystem` directly.

project test:

```sh
$ cat "test cases/frameworks/16 sdl2/"test.json 
```
```json
{
  "matrix": {
    "options": {
      "method": [
        { "val": "auto" },
        { "val": "pkg-config" },
        { "val": "config-tool" },
        { "val": "sdlconfig" },
        { "val": "extraframework", "expect_skip_on_os": ["!darwin"], "expect_skip_on_jobname": ["macos"] },
        { "val": "cmake", "expect_skip_on_jobname": ["bionic"] }
      ]
    }
  },
  "expect_skip_on_jobname": ["azure", "cygwin", "msys2"]
}
```

Yes, it is possible to have a project test target a specific dependency type. :)

Reply:
Thanks. That's what I needed. I updated the test. I also tested if it fails when `VULKAN_SDK` is invalid (a relative path).